jobs_detector/main.py

```diff
@@ -20,14 +31,133 @@ def jobs_detector():
 @click.option('-c', '--combinations', type=str,
               callback=lambda _, x: x.split(',') if x else x)
 def hacker_news(post_id, keywords, combinations):
-    """
-    This subcommand aims to get jobs statistics by parsing "who is hiring?"
-    HackerNews posts based on given set of keywords.
-    """
-    # HINT: You will probably want to use the `BeautifulSoup` tool to
-    # parse the HTML content of the website
-    pass
+    
+    data_to_parse = request_jobs_page(post_id)
+    job_posts = retrieve_job_postings(data_to_parse) # get all top comments
+    
+
+    # renaming this from response, I found it confusing because we already use response as a variable in the get URL and data request.
+    job_stats = []
+    
+    job_stats.append(get_total_jobs(job_posts))
+    
+    keyword_stats = get_keyword_stats(keywords, job_posts)
+    job_stats += keyword_stats_repr(keyword_stats, job_posts)
+
+    if combinations:
+        combined_stats = get_combination_stats(combinations, job_posts)
+        job_stats += combinations_stats_repr(combined_stats, job_posts)
+    
+    print(job_stats)
+    return job_stats
+    
+
+def get_keyword_stats (keywords, job_posts) :
+    ''' returns a dict with key = keyword, value = count of keywords in job_posts  
+        "keywords": list of keywords to search the comments for.
+            eq: ['Django', 'Python']
+        "job_posts": List of comments (top comments) in HN thread
+    '''
+  
+    # initialize 
+    keyword_dict = defaultdict(int)
+
+    # clean up keywords, you only need lower case
+    formatted_keywords = [str(keyword).lower().strip() for keyword in keywords.split(",")]
```

Comment:
This might break in Python 2. You're coercing the type to a ASCII str. What happens if the keyword is `😱`?